pkg/3scale/amp/component/apicast_monitoring.go

```diff
@@ -0,0 +1,198 @@
+		},
+	}
+}
+
+func (apicast *Apicast) ApicastStagingMonitoringService() *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "apicast-staging-metrics",
+			Labels: apicast.Options.StagingMonitoringLabels,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Name:       "metrics",
+					Protocol:   v1.ProtocolTCP,
+					Port:       9421,
+					TargetPort: intstr.FromInt(9421),
+				},
+			},
+			Selector: map[string]string{"deploymentConfig": "apicast-staging"},
+		},
+	}
+}
+
+func (apicast *Apicast) ApicastProductionServiceMonitor() *monitoringv1.ServiceMonitor {
+	return &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "apicast-production",
+			Labels: apicast.Options.ProductionMonitoringLabels,
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{{
+				Port:   "metrics",
+				Path:   "/metrics",
+				Scheme: "http",
+			}},
+			Selector: metav1.LabelSelector{
+				MatchLabels: apicast.Options.ProductionMonitoringLabels,
+			},
+		},
+	}
+}
+
+func (apicast *Apicast) ApicastStagingServiceMonitor() *monitoringv1.ServiceMonitor {
+	return &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "apicast-staging",
+			Labels: apicast.Options.StagingMonitoringLabels,
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{{
+				Port:   "metrics",
+				Path:   "/metrics",
+				Scheme: "http",
+			}},
+			Selector: metav1.LabelSelector{
+				MatchLabels: apicast.Options.StagingMonitoringLabels,
+			},
+		},
+	}
+}
+
+func ApicastMainAppGrafanaDashboard(ns string) *grafanav1alpha1.GrafanaDashboard {
+	data := &struct {
+		Namespace string
+	}{
+		ns,
+	}
+	return &grafanav1alpha1.GrafanaDashboard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "apicast-mainapp",
+			Labels: map[string]string{
+				"monitoring-key": common.MonitoringKey,
+			},
+		},
+		Spec: grafanav1alpha1.GrafanaDashboardSpec{
+			Json: assets.TemplateAsset("monitoring/apicast-grafana-dashboard-1.json.tpl", data),
+			Name: fmt.Sprintf("%s/apicast-grafana-dashboard-1.json", ns),
+		},
+	}
+}
+
+func ApicastServicesGrafanaDashboard(ns string) *grafanav1alpha1.GrafanaDashboard {
+	data := &struct {
+		Namespace string
+	}{
+		ns,
+	}
+	return &grafanav1alpha1.GrafanaDashboard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "apicast-services",
+			Labels: map[string]string{
+				"monitoring-key": common.MonitoringKey,
+			},
+		},
+		Spec: grafanav1alpha1.GrafanaDashboardSpec{
+			Json: assets.TemplateAsset("monitoring/apicast-grafana-dashboard-2.json.tpl", data),
+			Name: fmt.Sprintf("%s/apicast-grafana-dashboard-2.json", ns),
+		},
+	}
+}
+
+func ApicastPrometheusRules(ns string) *monitoringv1.PrometheusRule {
+	return &monitoringv1.PrometheusRule{
+		ObjectMeta: v12.ObjectMeta{
```

Comment:
Same issue as in comment https://github.com/3scale/3scale-operator/pull/333#discussion_r443666233. I've also observed it in other *_monitoring.go files